### PR TITLE
🦺 [Tech Debt] Changed password criteria

### PR DESCRIPTION
**Pull request by kcw-grunt**
## Why
Partner passed the password criteria and it needed to match the Dashboard criteria

## Commits
- 1 Capital letter, 1 number, 8 min chars, accepts special chars
- Added tests

### CODE REVIEW

**Review by mosadialiou — CHANGES_REQUESTED** [review]
The code LGTM but to avoid any confusion we have to tell the user that the password has to be `8 chars minimum, 1 Capital letter and 1 number, accepts special characters`

**Comment by kcw-grunt** [comment]
> The code LGTM but to avoid any confusion we have to tell the user that the password has to be `8 chars minimum, 1 Capital letter and 1 number, accepts special characters`

Agreed! It only took 1 year to get the info.  I wrote an issue for an improvement: #260

File: loafwallet/RegistrationViewModel.swift [hunk 1]
```diff
@@ -164,15 +164,15 @@ class RegistrationViewModel: ObservableObject {
...
     /// - Returns: Bool
     func isPasswordValid(passwordString: String) -> Bool {
         
-        guard passwordString.count >= 6 else {
+        guard passwordString.count >= 8 else {
             return false
         } 
         
-        if try! NSRegularExpression(pattern: "^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{6,}$", options: .caseInsensitive)
+        if try! NSRegularExpression(pattern: "^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d$&+,:;=?@#|'<>.^*()%!-]{8,}$", options: .caseInsensitive)
```
Comment (mosadialiou) [inline loafwallet/RegistrationViewModel.swift:175]: @kcw-grunt revert this change

File: loafwallet/RegistrationViewModel.swift [hunk 2]
```diff
@@ -164,15 +164,15 @@ class RegistrationViewModel: ObservableObject {
     }
     
     /// Password  Validator
-    /// - Parameter passwordString: 6 chars minimum
+    /// - Parameter passwordString: 8 chars minimum, 1 Capital letter and 1 number, accepts special characters
     /// - Returns: Bool
     func isPasswordValid(passwordString: String) -> Bool {
         
-        guard passwordString.count >= 6 else {
+        guard passwordString.count >= 8 else {
```
Comment (mosadialiou) [inline loafwallet/RegistrationViewModel.swift:171]: @kcw-grunt revert this change for now

File: loafwallet/RegistrationViewModel.swift [hunk 3]
```diff
@@ -164,15 +164,15 @@ class RegistrationViewModel: ObservableObject {
     }
     
     /// Password  Validator
-    /// - Parameter passwordString: 6 chars minimum
+    /// - Parameter passwordString: 8 chars minimum, 1 Capital letter and 1 number, accepts special characters
```
Comment (mosadialiou) [inline loafwallet/RegistrationViewModel.swift:167]: @kcw-grunt revert this change

File: loafwalletTests/Class Tests/LitecoinCardTests/RegistrationViewModelTests.swift [hunk 4]
```diff
@@ -69,13 +68,42 @@ class RegistrationViewModelTests: XCTestCase {
          
     }
     
-    func testIsPasswordDataValid() throws {
+    func testIsPasswordValid() throws {
```
Comment (mosadialiou) [inline loafwalletTests/Class Tests/LitecoinCardTests/RegistrationViewModelTests.swift:71]: @kcw-grunt revert all changes in this file

**Review by mosadialiou — CHANGES_REQUESTED** [review]

File: loafwallet/RegistrationViewModel.swift (hunk 2 above)
Comment (kcw-grunt) [reply loafwallet/RegistrationViewModel.swift:171]: oh yeah...good point.

File: loafwalletTests/Class Tests/LitecoinCardTests/RegistrationViewModelTests.swift (hunk 4 above)
Comment (mosadialiou) [reply loafwalletTests/Class Tests/LitecoinCardTests/RegistrationViewModelTests.swift:71]: @kcw-grunt you didn't revert the changes! put the file back as it was before the changes

**Review by mosadialiou — CHANGES_REQUESTED** [review]

**Review by mosadialiou — APPROVED** [review]
LGTM

**Pull request merged**